File: odroid/Share/src/bert/share/common/BottleConstants.java
/**
 * Copyright 2018-2019. Charles Coughlin. All Rights Reserved.
 *                 MIT License.
 */

package bert.share.common;


/**
 *  Define strings used in requests and responses between server and client.
 */
public interface BottleConstants   {   
	
	// Well-known keys for the properties map inside a request/response
	// Additional data values as appropriate are keyed with the JointProperty keys
	public final static String APPENDAGE_NAME      = "appendage";  // Request applies to this appendage, value is a Appendage
	public final static String COMMAND_NAME        = "command";    // Value is a well-known command name, see below
	public final static String ERROR               = "error";      // Request resulted in an error, value is error text
	public final static String JOINT_NAME          = "joint";      // Request applies to this joint, value is a Joint
	public final static String LIMB_NAME           = "limb";       // Request applies to this limb, value is a Limb
	public final static String METRIC_NAME         = "metric";     // Value is a MetricType
	public final static String POSE_NAME           = "pose";       // Name of a pose, must exist in database
	public final static String PROPERTY_NAME       = "property";   // Value is a JointProperty. Subject of original request.
	public final static String TYPE                = "type";       // Type of request, a RequestType
	public final static String SOURCE              = "source";     // Original source of request, value is a HandlerType
	public final static String TEXT                = "text";       // End-user appropriate text result
	
	// Command names
	public final static String COMMAND_FREEZE      = "freeze";
	public final static String COMMAND_HALT        = "halt";
	public final static String COMMAND_RELAX       = "relax";
	public final static String COMMAND_RESET       = "reset";
	public final static String COMMAND_SHUTDOWN    = "shutdown";
	public final static String COMMAND_WAKE        = "wake";
	
	// Pose names (these are required to exist)
	public final static String POSE_HOME           = "home";
	public final static String POSE_NORMAL_SPEED   = "normal speed";
	
	// Message from tablet
	public final static int HEADER_LENGTH = 4;     // Includes semi-colon
	
	// Nominal values for torque and speed in percent
	public final static int SPEED_NORMAL         = 20;
	public final static int TORQUE_NORMAL        = 20;
	// For values that are boolean. Use these strings for "values"
	public final static String ON_VALUE       = "1";
	public final static String OFF_VALUE      = "0";

}

File: Bert/common/src/main/kotlin/chuckcoughlin/bert/common/BottleConstants.kt
/**
 * Copyright 2022. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.common

/**
 * Define strings used in requests and responses between server and client.
 */
object BottleConstants {
    // Well-known keys for the properties map inside a request/response
    // Additional data values as appropriate are keyed with the JointProperty keys
    const val APPENDAGE_NAME = "appendage" // Request applies to this appendage, value is a Appendage
    const val COMMAND_NAME = "command" // Value is a well-known command name, see below
    const val CONTROLLER_NAME = "controller" // Message needs to be addressed to a specific controller
    const val ERROR = "error" // Request resulted in an error, value is error text
    const val JOINT_NAME = "joint" // Request applies to this joint, value is a Joint
    const val LIMB_NAME = "limb" // Request applies to this limb, value is a Limb
    const val METRIC_NAME = "metric" // Value is a MetricType
    const val POSE_NAME = "pose" // Name of a pose, must exist in database
    const val PROPERTY_NAME = "property" // Value is a JointProperty. Subject of original request.
    const val TYPE = "type" // Type of request, a RequestType
    const val SOURCE = "source" // Original source of request, value is a HandlerType
    const val TEXT = "text" // End-user appropriate text result

    // Command names
    const val COMMAND_FREEZE = "freeze"
    const val COMMAND_HALT = "halt"
    const val COMMAND_RELAX = "relax"
    const val COMMAND_RESET = "reset"
    const val COMMAND_SHUTDOWN = "shutdown"
    const val COMMAND_SLEEP = "sleep"
    const val COMMAND_WAKE = "wake"

    // Controller names
    const val CONTROLLER_LOWER = "lower"
    const val CONTROLLER_UPPER = "upper"

    // Pose names (these are required to exist)
    const val POSE_HOME = "home"
    const val POSE_NORMAL_SPEED = "normal speed"

    // Message from tablet
    const val HEADER_LENGTH = 4 // Includes semi-colon

    // Nominal values for torque and speed in percent
    const val SPEED_NORMAL = 20
    const val TORQUE_NORMAL = 20

    // For values that are boolean. Use these strings for "values"
    const val ON_VALUE = "1"
    const val OFF_VALUE = "0"

    // Default responses
    const val NO_ERROR = "No Error"
    const val NO_SOURCE = "No Source"

}